### PR TITLE
Feature suggestion: Add CC License to images

### LINKED ISSUE #71: [FEATURE] Add possibility to set a license
### Detailed description of the problem

It should be possible that you can select a license e.g. [Creative Commons](https://creativecommons.org/) in the global settings for your images. And that you can change it for individual images, but by default select the global one.

### PR DESCRIPTION
**Pull request by bennettscience**
Fixes #71 

This attempts to implement Issue #71 and allow the user to add a Creative Commons license to a photo. 

`Photo::setLicense` was added to the Photo module to facilitate storing a string-based license. It takes input from a modal window triggered in the UI.

### CODE REVIEW

File: php/database/update_030205.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+/**
+ * Update to version 3.2.4
```
Comment (ildyria) [inline php/database/update_030205.php]: This should be 3.2.5

File: php/database/update_030205.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// Add small to photos
+$query  = Database::prepare($connection, "SELECT `license` FROM `?` LIMIT 1", array(LYCHEE_TABLE_PHOTOS));
+$result = Database::execute($connection, $query, 'update_030205', __LINE__);
+
+if ($result===false) {
+
+	$query  = Database::prepare($connection, "ALTER TABLE `?` ADD `license` varchar(100) NOT NULL DEFAULT", array(LYCHEE_TABLE_PHOTOS));
+	$result = Database::execute($connection, $query, 'update_030205', __LINE__);
+
+	if ($result===false) Response::error('Could not add licnese-field to database!');
```
Comment (ildyria) [inline php/database/update_030205.php:19]: typo here.

File: php/database/update_030205.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+use Lychee\Modules\Database;
+use Lychee\Modules\Response;
+
+// Add small to photos
+$query  = Database::prepare($connection, "SELECT `license` FROM `?` LIMIT 1", array(LYCHEE_TABLE_PHOTOS));
+$result = Database::execute($connection, $query, 'update_030205', __LINE__);
+
+if ($result===false) {
+
+	$query  = Database::prepare($connection, "ALTER TABLE `?` ADD `license` varchar(100) NOT NULL DEFAULT", array(LYCHEE_TABLE_PHOTOS));
```
Comment (ildyria) [inline php/database/update_030205.php]: varchar(20) is enough. :)

Comment (ildyria) [reply php/database/update_030205.php]: also seems to miss a space after `DEFAULT` and before `''`

File: npm-debug.log [hunk 4]
```diff
@@ -0,0 +1,24 @@
+0 info it worked if it ends with ok
```
Comment (ildyria) [inline npm-debug.log:1]: Delete this file.

File: php/Modules/Photo.php [hunk 5]
```diff
@@ -1148,6 +1149,30 @@ public function setDescription($description) {
...
+	 * @return boolean Returns true when successful.
+	 */
+	public function setLicense($license) {
+
+		// Check dependencies
+		Validator::required(isset($this->photoIDs), __METHOD__);
+
+		// Call plugins
+		Plugins::get()->activate(__METHOD__, 0, func_get_args());
+
```
Comment (ildyria) [inline php/Modules/Photo.php:1165]: You need to sanitize your licence here:
```php
$licenses = [ '', 'CC0', 'CC-BY', 'CC-BY-ND', 'CC-BY-SA', 'CC-BY-ND', 'CC-BY-NC-ND', 'CC-BY-SA'];
$found = false;
$i = 0;
while(!$found && $i < count($licenses))
{
	if ($licenses[$i] == $license) $found = true;
	$i++;
}
if(!$found)
{
	// you might want to log this.
	return false;
}
```

**Review by ildyria — CHANGES_REQUESTED** [review]
You are missing the license information in the `Photo::PrepareData` function.

File: php/database/update_030205.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+use Lychee\Modules\Database;
+use Lychee\Modules\Response;
+
+// Add small to photos
+$query  = Database::prepare($connection, "SELECT `license` FROM `?` LIMIT 1", array(LYCHEE_TABLE_PHOTOS));
+$result = Database::execute($connection, $query, 'update_030205', __LINE__);
+
+if ($result===false) {
+
+	$query  = Database::prepare($connection, "ALTER TABLE `?` ADD `license` varchar(20)", array(LYCHEE_TABLE_PHOTOS));
```
Comment (ildyria) [inline php/database/update_030205.php]: after `varchar(20)` and before `''` please add ` NOT NULL DEFAULT `.
result should be:
```php
$query  = Database::prepare($connection, "ALTER TABLE `?` ADD `license` varchar(20) NOT NULL DEFAULT ", array(LYCHEE_TABLE_PHOTOS));

**Review by ildyria — CHANGES_REQUESTED** [review]
Default string is not `null` it is an empty string: `''`

**Comment by ildyria** [comment]
Also can you update `Locale/English.php` with the same localization that you added to `lychee_locale.js` ?
It is optional, but it would help those who use English.php as a base for translation. :+1:

**Comment by bennettscience** [comment]
Sorry, I messed up the commit tree, but the requested changes have been added to the PR.

**Comment by ildyria** [comment]
No worries with the commit tree, I usually Squash and merge. :)

**Pull request merged**